Sources/DeepLearning/Layer.swift

```diff
@@ -1295,3 +1295,59 @@ public struct Reshape<Scalar: TensorFlowFloatingPoint>: Layer {
         return input.reshaped(toShape: shape)
     }
 }
+
+struct LSTMCell<Scalar: TensorFlowFloatingPoint>: Layer {
+
+    var inputWeight: Tensor<Scalar>
+    var updateWeight: Tensor<Scalar>
+    var forgetWeight: Tensor<Scalar>
+    var forgetBias: Tensor<Scalar>
+    var outputWeight: Tensor<Scalar>
+
+    init(inputSize: Int32, hiddenSize: Int32) {
+        let concatenatedInputSize = inputSize + hiddenSize
+        self.inputWeight = Tensor<Scalar>(glorotUniform: [concatenatedInputSize, hiddenSize])
+        self.updateWeight = Tensor<Scalar>(glorotUniform: [concatenatedInputSize, hiddenSize])
+        self.forgetWeight = Tensor<Scalar>(glorotUniform: [concatenatedInputSize, hiddenSize])
+        self.forgetBias = Tensor<Scalar>(zeros: [hiddenSize])
+        self.outputWeight = Tensor<Scalar>(glorotUniform: [concatenatedInputSize, hiddenSize])
+    }
+
+    struct HiddenState: Differentiable {
```

Comment:
Why call it “HiddenState” instead of the existing associated type name “Output”? This way, protocol requirement “Output” is going to be inferred as a typealias of “HiddenState”, which is confusing.

Reply:
Because the output _is_ is the hidden state - like for example, in the Input struct, we've got `var hidden: HiddenState` which represents the previous timestep's output (which is the hidden state). Wdyt?